site/content/entry/citra-progress-report-2018-Q1.md

```diff
@@ -0,0 +1,233 @@
+between different ways to deal with this issue, with different drawbacks:
+
+1. Wait for the GPU to finish and copy to the destination buffer. Usually this method is the fastest
+for big uploads, fine on common gaming, but terrible for emulators.
+2. Allocate a new buffer and tell the GPU which new buffer shall be used now. You also need to care
+about freeing the old buffer once it isn't in use any more. This tends to have a huge CPU overhead and
+isn't available on partial buffer updates such as Lighting LUTs.
+3. Copy the data to an internal staging buffer and ask the GPU to copy it again to the destination
+when it is ready. It uses twice the memory bandwidth, and switching between rendering and copying also
+takes a while for the GPU. This proves to be a significant bottleneck if done for every single draw call.
+
+In the end, all of these are bad in terms of performance. It is a simple upload call, but it either
+requires many copies and context switches, or the GPU is stalled (_More on that in [here](https://de.slideshare.net/CassEveritt/approaching-zero-driver-overhead)_). Instead, Citra
+allocates a big staging buffer, copies the data to the current position within the buffer,
+and tells the GPU to directly access the data from there. The driver has no scope of interpreting what
+we ask it to do.
+
+_This has a downside as well_. The GPU now accesses the main memory over PCIe. So if we're going to
+reuse the same data in the next frame, it has to go through the PCIe bottleneck again. But, we never
+use the same data again and hence gain a big performance boost.
+
+[wwylele](https://github.com/wwylele) tweaked the configuration UI and renamed some parts of it.
+However, this caused some confusion about the configuration window and GPU features.
+
+### What's changed in the UI ?
+
+With the initial release of GLvtx, we added support for running shaders on the host GPU. So, we had
+a dropdown menu called **Shader Emulation** to select either CPU or GPU, to run the shaders. The
+option has now been renamed as **Enable Hardware Shader**. When enabled, it makes use of the host
+GPU, and when disabled, it uses the CPU alone.
+
+Due to many of the users having older hardware, we introduced an option called **Accurate Hardware
+Shader**. It was later renamed to **Accurate Multiplication**, because technically that's what it was
+doing. ***When enabled it would render games with more accuracy at the cost of slower performance***.
+Same goes for **Accurate Geometry Shader**. Earlier it was configurable via the ini file, but now
+it's a check box to enable or disable accurate geometry shader.
+
+&nbsp;
+{{< sidebyside "image" "/images/entry/citra-progress-report-2018-q1/"
+    "renderer-settings.png=Old Graphics Tab"
+    "renderer-settings-new.png=New Graphics Tab" >}}
+
+During it's time in Canary, a lot of testing was done by our contributors. They tested many number of
```

Comment:
Grammar: Instead of "They tested many number of games...", "They tested many games..."